packages/@aws-cdk/aws-cloudformation/lib/pipeline-actions.ts

```diff
@@ -459,8 +459,8 @@ class SingletonPolicy extends cdk.Construct {
 
   private statements: { [key: string]: iam.PolicyStatement } = {};
 
-  private constructor(private readonly role: iam.Role) {
-    super(role, SingletonPolicy.UUID);
+  private constructor(private readonly role: iam.IRole) {
+    super(role as unknown as cdk.Construct, SingletonPolicy.UUID);
```

Comment:
Use a feature test? `cdk.Construct.isConstruct()` should exist (if not you can make it).